pymc_extras/statespace/models/structural/components/cycle.py

```diff
@@ -0,0 +1,307 @@
+import numpy as np
+
+from pytensor import tensor as pt
+from pytensor.tensor.slinalg import block_diag
+from scipy import linalg
```

Comment:
Don't use `scipy.linag.block_diag`, only `pt.linalg.block_diag`. Otherwise we can't reason about the block_diag in future rewrites.